specification/cognitiveservices/AnomalyDetector/main.tsp

```diff
@@ -10,7 +10,7 @@ using TypeSpec.Versioning;
 
 @versioned(APIVersion)
 @service({title: "Anomaly Detector"})
-@useAuth(ApiKeyAuth<ApiKeyLocation.header, "Ocp-Apim-Subscription-Key">)
+@useAuth(AnomalyDetectorApiKeyAuth)
```

Comment:
I would use [service traits](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAYXp1cmUtdG9vbHMvdHlwZXNwZWMtxhVjb3JlIjsKCnVzaW5nIFR5cGVTcGVjLkh0dHA70BVSZXN0yBVBxD4uQ29yZTsKCkBzZXJ2aWNlKHsgdGl0bGU6ICJEb2VzIGl0IHdvcms%2FIn0pxiRlcigie2VuZHBvaW50fS9hbm9tYWx5ZGV0ZWN0b3IiLCAKICAiU29tZXRoxGVzaccKIiwKICB7CiAgyDw6IHN0cmluZzsKfSkKbmFtZXNwYWNlIElUb2xkWW91U287Cgptb2RlbCBBcGlWZXJzaW9uxUFAc2VnbWVudCgiIinED3BhdGgoImFwaS12xibGF2RvYygiVGhlIEFQSSDHGiB0byB1c2UgZm9yIHRoaXMgb3BlcmF0aW9uLsU1Yclq6wCeCgphbGlhc%2BsAh1BhdGhQYXJhbWV0ZXIgPSBUcmFpdHMuxQdPdmVycmlkZTzHFckzxy%2FFFzzKTD4%2BOwoKaW50ZXJm5AD%2FT%2BgAkXMgZXh0ZW5kcyBSZXNvdXJjZcobyz%2FuAIsm6ACLTm9Db25kacQuYWxSZXF1ZXN0c8wfUmVwZWF0YWJsZdIeU3Vw5AJ2c0NsaWVudMcgSWQ%2BIHsK5AEOQOQCRS5y5wCWKCLlAd9zIinnAa5UxA%2FmAalrZXnEB3Zpc2liaWxpdHkoInJlYWTlAWvFNklk7QFob3AgxCDGQmlz6wEN5AK5xW9SZWFkPMUhPjvEMHdyaXRl3DFDcmVhdGVPclJlcGxhY2XIPA%3D%3D)

Reply:
Service traits will move the api-version from the x-ms-parameterized host to the method and I think we want to avoid that.